### DISCUSSION TITLE
LocalContext/ClientContext

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#3938] [Questions] [answered]
I'm migrating to Csla 7 and I'm not sure my understanding of LocalContext/ClientContext is correct anymore.  Is it safe to store things there so they are accessible on the client?  If I put something in LocalContext is it available to all future data portal calls?  How does async and threads affect things?

**Comment by rockfordlhotka** [comment]
Local/Client context should be valid in CSLA 7 - as long as you aren't using Blazor 8.

The data portal flows ClientContext from client to server. LocalContext is local to a specific device and doesn't flow through the data portal.

**Reply by rockfordlhotka** [reply, marked as answer]
If you actually create threads (like Task.Run), it is up to you to flow any context to the new thread.

Normal async/await constructs don't actually create threads, so generally no extra work is required. But Task.Run (or _shudder_ `new Thread` are a whole other matter).

**Reply by ajohnstone-ks** [reply]
@rockfordlhotka Thank you!  Yes we still do have some places where threads are manually created (the code base dates back to 2004 or so) so I will be on the lookout for that... although I suspect using the static context manager would solve this problem in a desktop client app?
